### PR TITLE
Add a pretty-printer for the Cimple+Apidsl AST.

### PR DESCRIPTION
**Pull request by iphydf**
This change is [Reviewable](https://reviewable.io/reviews/toktok/hs-cimple/2)

### CODE REVIEW

File: src/Language/Cimple/Pretty.hs [hunk 1]
```diff
@@ -0,0 +1,574 @@
...
+ppStmt :: Node (Lexeme Text) -> Doc
+ppStmt = ppDecl
+
+ppDeclList :: [Node (Lexeme Text)] -> Doc
+ppDeclList = ppLineSep ppDecl
+
+ppDecl :: Node (Lexeme Text) -> Doc
+ppDecl decl = case decl of
+    PreprocElif cond decls (PreprocElse []) ->
+        text "#esif" <+> ppExpr cond <$>
```
Comment (robinlinden) [inline src/Language/Cimple/Pretty.hs]: esif?

File: test/Language/Cimple/PrettySpec.hs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            compact "/* foo.bar,baz-blep */"
+                >>= (`shouldBe` "/* foo. bar, baz - blep\n */\n")
+            compact "/* foo? */" >>= (`shouldBe` "/* foo ?\n */\n")
+
+        it "formats number ranges and negative numbers without spacing" $ do
+            compact "/* 123 - 456 */" >>= (`shouldBe` "/* 123-456\n */\n")
+            compact "/* - 3 */" >>= (`shouldBe` "/* -3\n */\n")
+            compact "/* a-b */" >>= (`shouldBe` "/* a - b\n */\n")
+
+        it "formats pointer types with east-const" $ do
```
Comment (robinlinden) [inline test/Language/Cimple/PrettySpec.hs:78]: East const best const?

File: src/Language/Cimple/Pretty.hs (hunk 1 above)
Comment (iphydf) [reply src/Language/Cimple/Pretty.hs]: Fixed.

File: test/Language/Cimple/PrettySpec.hs (hunk 2 above)
Comment (iphydf) [reply test/Language/Cimple/PrettySpec.hs:78]: It's mostly just easier to pretty-print, but yes :).

**Review by robinlinden — APPROVED** [review]

**Pull request merged**